rts/System/GlobalRNG.h

```diff
@@ -128,8 +128,8 @@ template<typename RNG, bool synced> class CGlobalRNG {
 	rng_res_type NextInt(rng_res_type N = max()) { return ((*this)(N)); }
 
 	float NextFloat() { return (NextFloat01(1 << ndig())); }
-	float NextFloat01(rng_res_type N) { return ((NextInt(N) * 1.0f) / N); } // [0,1) rounded to multiple of 1/N
-	float NextFloat24() { return (math::ldexp(NextInt(1 << ndig()), -ndig())); } // [0,1) rounded to multiple of 1/(2^#digits)
+	float NextFloat01(rng_res_type N) { return static_cast<float>(NextInt(N)) / N; } // [0,1) rounded to multiple of 1/N
```

Comment:
1) multiplying `T` by `1.0f` promotes `T` to a float, which has the same effect as `static_cast`'ing it (but is shorter)
2) similarly `ldexp` implicitly converts its argument to a float (`streflop::Simple` technically), why the cast?
3) not to be blunt, but why are you touching RNG code in the middle of a CEG-related commit bundle?